tornado/ioloop.py

```diff
@@ -616,10 +616,16 @@ def _run_callback(self, callback):
                     # result, which should just be ignored.
                     pass
                 else:
-                    self.add_future(ret, lambda f: f.result())
+                    self.add_future(ret, lambda f: self._handle_exc(f))
         except Exception:
             self.handle_callback_exception(callback)
 
+    def _handle_exc(self, future):
```

Comment:
This method could use a different name to distinguish it from the adjacent `handle_callback_exception`. Maybe `_handle_future_result`?

Or I don't think it even needs the exception handling here. Just retrieve `future.result()` and discard it if no exception is raised (so call it `_discard_future_result`?). I think this will be called from a `_run_callback` somewhere so the exception will be logged when it escapes.

For maximum cleverness (and to avoid naming anything) I suppose you could do something like `lambda f: not f.result()` (on the assumption that `convert_yielded` will never decide to be clever with boolean values).